iothub/service/src/ServiceClient.cs

```diff
@@ -634,9 +635,24 @@ private static TimeSpan GetInvokeDeviceMethodOperationTimeout(CloudToDeviceMetho
             // for the Device to connect and send a response. We also need to take into account
             // the transmission time for the request send/receive
             var timeout = TimeSpan.FromSeconds(15); // For wire time
-            timeout += TimeSpan.FromSeconds(cloudToDeviceMethod.ConnectionTimeoutInSeconds ?? 0);
-            timeout += TimeSpan.FromSeconds(cloudToDeviceMethod.ResponseTimeoutInSeconds ?? 0);
-            return timeout <= s_defaultOperationTimeout ? s_defaultOperationTimeout : timeout;
+            var connectionTimeOut = TimeSpan.FromSeconds(cloudToDeviceMethod.ConnectionTimeoutInSeconds ?? 0);
+            var responseTimeOut = TimeSpan.FromSeconds(cloudToDeviceMethod.ResponseTimeoutInSeconds ?? 0);
+
+            ValidateTimeOut(connectionTimeOut);
+            ValidateTimeOut(responseTimeOut);
+
+            timeout += connectionTimeOut;
+            timeout += responseTimeOut;
+
+            return timeout;
+        }
+
+        private static void ValidateTimeOut(TimeSpan connectionTimeOut)
+        {
+            if (connectionTimeOut.Seconds < 0)
+            {
+                throw new ArgumentException("Negative timeout");
```

Comment:
This technically introduces a breaking change.

Reply:
true. Should I remove this and let the service deal with negative values?

Reply:
The user can already get an exception for invalid values (when too large, right)? So that seems like it would be fine, that is not a breaking change.